cql3/cql_statement.hh

```diff
@@ -54,6 +54,11 @@ public:
         return false;
     }
 
+    // Returns true for statements that needs guard to be taken before the execution
```

Comment:
I think this deserves a better comment. What does "needs guard" even means here (we usually use the word "guard" in the context of group0 to mean something else).
Other comments you added together with this function "Prevent race between drop/truncate and tablet migration", but this is only slightly clearer... Can you please explain better - here and/or in a commit message - why there is a "race" between drop/truncate and tablet migration? Why we need to solve this race instead of just allowing it? Moreover, how do we know that a tablet migration wouldn't take 10 minutes, and what exactly happens to the table drop while it is ongoing?

Frankly, it would be better if we could drop a table immediately and any ongoing processes on this table - be it compaction, repair, view building or tablet migrations - should be aborted when this fact is noticed. We don't delay table deletion while compaction, repair, or view building is happening on this table - so why start now with tablets?

Reply:
I'm backing away from this method.
So next version will probably won't have it.